From patchwork Mon Apr 15 19:11:20 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Nadav Amit <nadav.amit@gmail.com>
X-Patchwork-Id: 10901439
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id B114F1669
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon, 15 Apr 2019 19:16:06 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 982FD28468
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon, 15 Apr 2019 19:16:06 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 8C38228867; Mon, 15 Apr 2019 19:16:06 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 3AB1D28468
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon, 15 Apr 2019 19:16:06 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1731590AbfDOTNj (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Mon, 15 Apr 2019 15:13:39 -0400
Received: from mail-pg1-f193.google.com ([209.85.215.193]:38281 "EHLO
        mail-pg1-f193.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1731011AbfDOTNi (ORCPT <rfc822;kvm@vger.kernel.org>);
        Mon, 15 Apr 2019 15:13:38 -0400
Received: by mail-pg1-f193.google.com with SMTP id j26so9039705pgl.5
        for <kvm@vger.kernel.org>; Mon, 15 Apr 2019 12:13:38 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=from:to:cc:subject:date:message-id:mime-version
         :content-transfer-encoding;
        bh=x3kU2Zqx+2bTZbqNyx1i0Gj2jD48pawuaSSBMGeJS+s=;
        b=UDP1PJWoiZRa/jLrIqbmdt420V7YV+BeYzwSleGole/anAMh+mWouMjm2UnVXN429R
         3+st1HZTMW+nD0oocsFPI+s+SgyNDhYtVc8DXviVJNLyMeoVOehvBver8PPLrjbDX++l
         c93c8B9fzsLfdes2aBz+fkhOv1wcI2PzsckkoFF6BMtkCWahZ7yRkQbilnZ6+KyWDPtp
         cSw3D/YZcdDu0ojU1dr7KfGD2qgTCUKHwE1Mv5xHXRXwi25VLHoAx7Q2mMmX1JXKVm/k
         XyB31pE6rUQ7hImuBuWv1HfXS44+mLzsZFM6eOb7nG4gWMfwDxGmTDkS0F9DqccqhSqO
         jNTA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:mime-version
         :content-transfer-encoding;
        bh=x3kU2Zqx+2bTZbqNyx1i0Gj2jD48pawuaSSBMGeJS+s=;
        b=cjKRnIg3Bij5UhM1kDG+zIgM2SWbcwLHaMFA8vqjLj0fCa1U6DB0ETIimiSNKepwZ3
         q1pk2pSNbQn2MnwuPhSmdfPBxKhnwKUSxVsWobMtDXRV9+snmB2ocleGMF3nYRacI4Am
         ld6yQJS3r50/yfqAYC5s6PjcXAnOH3g2xHrkgRu5QNwjlHQ7Ja5Mb7Co65ozC7p1yLfF
         nEtazqXf45QGH0Bp/YieLXP4U/mxIPhcXl3exV9nA6FaStNPqVn0Im6xTAcNAv+O+0KV
         Kro15J6JP0V3TVevVSJqhHqtwldnX2Vivgdg4WHFgI4z5szq9DEHOrO8htKGhdx2hJ/0
         XDsw==
X-Gm-Message-State: APjAAAVdjdl3XA7YDo5v9JoGnt2clQ1zc3VMufAfR5VscS5R4+rXLuyV
        0lZsi+IV1/i6yzL8ldtS7As=
X-Google-Smtp-Source: 
 APXvYqzUtu68i6+wI/kRq0WAsNHrVzLCeIUe8kAeLJw3iC37NalU7W/Jbm2+CPmTCJPS6q4t7mLFjw==
X-Received: by 2002:a63:700f:: with SMTP id l15mr71477866pgc.3.1555355617747;
        Mon, 15 Apr 2019 12:13:37 -0700 (PDT)
Received: from htb-2n-eng-dhcp405.eng.vmware.com ([66.170.99.1])
        by smtp.gmail.com with ESMTPSA id
 a17sm87115924pfj.123.2019.04.15.12.13.36
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Mon, 15 Apr 2019 12:13:37 -0700 (PDT)
From: nadav.amit@gmail.com
To: Paolo Bonzini <pbonzini@redhat.com>
Cc: kvm@vger.kernel.org, Nadav Amit <nadav.amit@gmail.com>
Subject: [kvm-unit-tests PATCH] x86: Fix INIT during boot
Date: Mon, 15 Apr 2019 12:11:20 -0700
Message-Id: <20190415191120.9472-1-nadav.amit@gmail.com>
X-Mailer: git-send-email 2.19.1
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

From: Nadav Amit <nadav.amit@gmail.com>

INIT is a level event and the trigger mode should be marked as such. In
addition, the SDM says that INIT deassertion should specify the "all
including-self" shorthand.

Signed-off-by: Nadav Amit <nadav.amit@gmail.com>
---
 x86/cstart64.S | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/x86/cstart64.S b/x86/cstart64.S
index 21db10f..57c3552 100644
--- a/x86/cstart64.S
+++ b/x86/cstart64.S
@@ -284,8 +284,8 @@ smp_init:
 	mov $(sipi_end - sipi_entry), %rcx
 	rep/movsb
 	mov $APIC_DEFAULT_PHYS_BASE, %eax
-	movl $(APIC_DEST_ALLBUT | APIC_DEST_PHYSICAL | APIC_DM_INIT | APIC_INT_ASSERT), APIC_ICR(%rax)
-	movl $(APIC_DEST_ALLBUT | APIC_DEST_PHYSICAL | APIC_DM_INIT), APIC_ICR(%rax)
+	movl $(APIC_DEST_ALLBUT | APIC_DEST_PHYSICAL | APIC_DM_INIT | APIC_INT_ASSERT | APIC_INT_LEVELTRIG), APIC_ICR(%rax)
+	movl $(APIC_DEST_ALLINC | APIC_DEST_PHYSICAL | APIC_DM_INIT | APIC_INT_LEVELTRIG), APIC_ICR(%rax)
 	movl $(APIC_DEST_ALLBUT | APIC_DEST_PHYSICAL | APIC_DM_STARTUP), APIC_ICR(%rax)
 	call fwcfg_get_nb_cpus
 1:	pause
